### PR TITLE
Add guidance for optional meta packages

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
## Summary

This change adds guidance for optional meta packages.

### CODE REVIEW

File: docs/core/distribution-packaging.md [hunk 1]
```diff
@@ -166,6 +166,18 @@ The following lists the recommended packages:
   - **Version:** \<sdk version>
   - **Contains:** (15)
 
+The following 2 meta packages are optional. They do bring value for end users, as they abstract the top-level package (dotnet-sdk) which simplifies the installation of the full set of .NET packages. These meta packages would reference a specific .NET SDK version.
```
Comment (gewarren) [inline docs/core/distribution-packaging.md:169]:
```suggestion
The following two meta packages are optional. They bring value for end users in that they abstract the top-level package (dotnet-sdk), which simplifies the installation of the full set of .NET packages. These meta packages reference a specific .NET SDK version.
```

File: docs/core/distribution-packaging.md [hunk 2]
```diff
@@ -166,6 +166,18 @@ The following lists the recommended packages:
...
+
+- `dotnet[major]` - Installs the specific sdk version
```
Comment (gewarren) [inline docs/core/distribution-packaging.md:171]:
```suggestion
- `dotnet[major]` - Installs the specified SDK version
```

File: docs/core/distribution-packaging.md [hunk 3]
```diff
@@ -166,6 +166,18 @@ The following lists the recommended packages:
...
+  - **Version:** \<sdk version>
+  - **Example:** dotnet7
+  - **Dependencies:** `dotnet-sdk-[major].[minor]`
+
+- `dotnet` - Installs the specific sdk version determined by distros to be the primary, usually the latest available
```
Comment (gewarren) [inline docs/core/distribution-packaging.md:176]:
```suggestion
- `dotnet` - Installs a specific SDK version determined by distros to be the primary version&mdash;usually the latest available
```

**Review by gewarren — APPROVED** [review]
I left some minor suggestions.

File: docs/core/distribution-packaging.md (hunk 3 above)
Comment (gewarren) [reply docs/core/distribution-packaging.md:176]: Yes, they should be updated to SDK - thanks.

Comment (NikolaMilosavljevic) [reply docs/core/distribution-packaging.md:176]: Thanks! I'll commit your suggestions and make the other change.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**